common-lib/src/main/scala/com/gu/mediaservice/lib/ImageIngestOperations.scala

```diff
@@ -42,9 +46,29 @@ class ImageIngestOperations(imageBucket: String, thumbnailBucket: String, config
     storeImage(imageBucket, optimisedPngKeyFromId(storableImage.id), storableImage.file, Some(storableImage.mimeType),
       overwrite = true)
 
+  private def bulkDelete(bucket: String, keys: Seq[String]): Future[Map[String, Boolean]] = Future {
+    try {
+      client.deleteObjects(
+        new DeleteObjectsRequest(bucket).withKeys(keys: _*)
+      )
+      keys.map { key =>
+        key -> true
+      }.toMap
+    } catch {
+      case partialFailure: MultiObjectDeleteException =>
+        // FIXME log partial failure
+        keys.map { key =>
+          key -> partialFailure.getErrors.asScala.map(_.getKey).toList.contains(key)
+        }.toMap
```

Comment:
This looks like it returns key -> true for each key which appears in getErrors (so failed to delete) - but the success branch above returns key -> true for each key which is successfully deleted
```suggestion
        val errorKeys = partialFailure.getErrors.asScala.map(_.getKey).toSet
        keys.map { key =>
          key -> !errorKeys.contains(key)
        }.toMap
```